native-schema-registry/golang/pkg/gsrserde-go/helpers.go

```diff
@@ -0,0 +1,82 @@
+package gsrserde
+
+import (
+	"unsafe"
+
+	"github.com/awslabs/aws-glue-schema-registry/native-schema-registry/golang/pkg/GsrSerDe"
+)
+
+// createReadOnlyByteArray creates a SWIG Read_only_byte_array from Go byte slice
+func createReadOnlyByteArray(data []byte, err GsrSerDe.Glue_schema_registry_error) (GsrSerDe.Read_only_byte_array, error) {
+	if len(data) == 0 {
+		return nil, ErrEmptyData
+	}
+	
+	// Get pointer to first byte
+	dataPtr := (*byte)(unsafe.Pointer(&data[0]))
+	
+	// Create read only byte array
+	roba := GsrSerDe.NewRead_only_byte_array(dataPtr, int64(len(data)), err)
+	
+	// Check if error was set
+	if err != nil && err.Swigcptr() != 0 {
+		return nil, extractError("create read only byte array", err)
+	}
+	
+	return roba, nil
+}
+
+// mutableByteArrayToGoSlice converts a SWIG Mutable_byte_array to Go byte slice
+func mutableByteArrayToGoSlice(mba GsrSerDe.Mutable_byte_array) []byte {
+	if mba == nil || mba.Swigcptr() == 0 {
+		return nil
+	}
+	
+	// Get the data pointer and length
+	dataPtr := mba.Get_data()
+	maxLen := mba.Get_max_len()
+	
+	if dataPtr == nil || maxLen <= 0 {
+		return nil
+	}
+	
+	// Create a Go slice from the C memory
+	// We need to copy the data to ensure memory safety
+	result := make([]byte, maxLen)
+	
+	// Use unsafe to create a temporary slice view of the C memory
+	cSlice := (*[1 << 30]byte)(unsafe.Pointer(dataPtr))[:maxLen:maxLen]
```

Comment:
Minor: Hard to parse, please consider using variables. Adding a comment for future readers.